### DISCUSSION TITLE
Retry Logic

### DISCUSSION THREAD

**Discussion by fgarcia1685** [quartznet/quartznet#1813] [General]
Hello, I’m wondering what kind of document there is around retry logic within quartz. Trying to deal with occasion loss in connection and rescheduling. Usually by patching of SQL Server.

**Comment by lahma** [comment]
If you mean retry by Quartz for connectivity problems (vs you own job retrying), there's some retry logic in JobStoreSupport:

https://github.com/quartznet/quartznet/blob/e3c374b6e0dc5287ffef377dcc99dd0b58776a24/src/Quartz/Impl/AdoJobStore/JobStoreSupport.cs#L3572-L3610

**Reply by fgarcia1685** [reply]
I am not as familiar with how IsTransient works? Also I took a look at the code as we have the quartz project directly in our solution(for telemetry purposes). In that version I see it accounts for many SQL error codes and the ConnectionAndTransactionHolder rolls back if the IsTransient is true.

**Reply by lahma** [reply]
What is the question?
